daemon/SecretsImpl/secrets.cpp

```diff
@@ -681,26 +682,31 @@ bool Daemon::ApiImpl::SecretsRequestQueue::compareTestCipherText(
         return false;
     }
 
-    const QString lockCodeCheckFileName = m_autotestMode
+    const QString lockCodeCheckDirName = m_autotestMode
             ? QLatin1String("lockcodecheck-test")
             : QLatin1String("lockcodecheck");
-    const QString lockCodeCheckPath = secretsDir.absoluteFilePath(lockCodeCheckFileName);
-    if (!QFile::exists(lockCodeCheckPath)) {
+    const QString lockCodeCheckDirPath = secretsDir.absoluteFilePath(lockCodeCheckDirName);
+
+    QByteArray previousData;
+    DataProtector dataProtector(lockCodeCheckDirPath);
+    DataProtector::Status s = dataProtector.getData(&previousData);
+
+    if (s != DataProtector::Success) {
+        qCWarning(lcSailfishSecretsDaemon) << "Can't read lock code data, assuming it's corrupted. DataProtector returned:" << s;
+        qCWarning(lcSailfishSecretsDaemon) << "NOTE: If you are running a pre-release sailfish-secrets version, you need to delete all old data before proceeding";
+        abort();
```

Comment:
I'm not sure whether aborting secrets is good upon failure without letting user to know. How do you recover from the error?

Reply:
Let's go with this for now and improve later. Fyi @chriadam

Reply:
My suggestion:

* For 2.2.0 let's just go with `abort()`. Alternatives: automatically delete the data when it's corrupted, or just let the daemon run even though anything that used the lock code (or salt) data won't work anymore.
* For 2.2.1 and onwards: we should add a user-friendly dialog, maybe a system prompt, that lets the user know what happened and offer to delete all the corrupted data.

Reply:
For now I will add a `TODO` comment as a reminder to not forget this later.